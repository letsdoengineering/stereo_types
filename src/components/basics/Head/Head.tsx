import React from 'react'
import { Helmet } from 'react-helmet'

type HeadProps = {
  title: string
}

/** Head: reusable <head> elements built with react-helmet */
const Head: React.FC<HeadProps> = ({ title }: HeadProps) => {
  return (
    <Helmet
      htmlAttributes={{
        lang: 'en',
      }}
    >
      <title>{title}</title>
      <script src='https://kit.fontawesome.com/3fb4da1d2f.js' crossOrigin='anonymous' />
    </Helmet>
  )
}

export default Head
