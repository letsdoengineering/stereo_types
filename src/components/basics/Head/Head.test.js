import renderer from 'react-test-renderer'
import Head from './Head.tsx'
import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import * as React from 'react'

describe('Head', () => {
    it('has not changed the JSX layout of the page from previous snapshot', () => {
        const tree = renderer.create(<Head>Some header</Head>).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('renders the page DOM that is accessible', async () => {
        const { container } = render(<Head>Some header</Head>)
        const results = await axe(container)

        expect(results).toHaveNoViolations()
    })
})
