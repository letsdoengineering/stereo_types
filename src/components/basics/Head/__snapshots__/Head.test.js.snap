// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head has not changed the JSX layout of the page from previous snapshot 1`] = `null`;
